### PR TITLE
Update to CosmWasm v0.14.0

### PR DESCRIPTION
**Pull request by ethanfrey**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -906,10 +906,8 @@ jobs:
             done
       - run:
           name: Install check_contract
-          # Uses --debug for compilation speed
-          # FIXME: Change when `check_contract` (part of `cosmwasm-0.14.0`) is published
-          #command: cargo install --debug --features iterator --example check_contract -- cosmwasm-vm
-          command: cargo install --debug --features iterator --git https://github.com/CosmWasm/cosmwasm --tag=v0.14.0-beta5 --example check_contract -- cosmwasm-vm
+          # Uses --debug for compilation speed (assumes 0.14.0 version of check_contract)
+          command: cargo install --debug --features iterator --example check_contract -- cosmwasm-vm
```
Comment (webmaster128) [inline .circleci/config.yml:910]: Does this work? You can use it for pre-releases that are on crated.io as well

```suggestion
          command: cargo install --debug --features iterator --example check_contract --version 0.14.0 -- cosmwasm-vm
```

File: contracts/cw20-escrow/src/state.rs [hunk 2]
```diff
@@ -77,7 +77,7 @@ impl Escrow {
         }
 
         if let Some(end_time) = self.end_time {
-            if env.block.time > end_time {
+            if env.block.time > Timestamp::from_seconds(end_time) {
```
Comment (webmaster128) [inline contracts/cw20-escrow/src/state.rs:80]: You could turn the state into `pub end_time: Option<Timestamp>` now if you want

File: contracts/cw20-ics20/src/ibc.rs [hunk 3]
```diff
@@ -442,8 +441,7 @@ mod test {
                 channel_id: my_channel.to_string(),
             },
             sequence: 3,
-            timeout_block: None,
-            timeout_timestamp: Some(1665321069000000000u64),
+            timeout: IbcTimeout::with_timestamp(Timestamp::from_seconds(1665321069)),
```
Comment (webmaster128) [inline contracts/cw20-ics20/src/ibc.rs:444]: If you like it short:

```suggestion
            timeout: Timestamp::from_seconds(1665321069).into(),
```

Not sure this improves the code though

File: contracts/cw3-flex-multisig/src/contract.rs [hunk 4]
```diff
@@ -804,7 +804,14 @@ mod tests {
...
         move |block: &mut BlockInfo| {
             match voting_period {
-                Duration::Time(duration) => block.time -= duration,
+                Duration::Time(duration) => {
+                    let (secs, nanos) = (
+                        block.time.nanos() / 1_000_000_000,
+                        block.time.nanos() % 1_000_000_000,
+                    );
+                    let earlier = Timestamp::from_seconds(secs - duration);
+                    block.time = earlier.plus_nanos(nanos);
```
Comment (webmaster128) [inline contracts/cw3-flex-multisig/src/contract.rs:813]:
```suggestion
                    block.time = Timestamp::from_nanos(block.time.nanos() - duration * 1_000_000_000);
```

File: packages/cw0/src/expiration.rs [hunk 5]
```diff
@@ -24,7 +24,10 @@ impl fmt::Display for Expiration {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
             Expiration::AtHeight(height) => write!(f, "expiration height: {}", height),
-            Expiration::AtTime(time) => write!(f, "expiration time: {}", time),
+            Expiration::AtTime(time) => {
+                let secs = time.nanos() / 1_000_000_000;
+                write!(f, "expiration time: {}", secs)
```
Comment (webmaster128) [inline packages/cw0/src/expiration.rs:29]: `Timestamp` has Display now if you want to use it here

**Review by webmaster128 — APPROVED** [review]
Looks good. Some potential improvements

**Comment by ethanfrey** [comment]
Thanks for the review

File: contracts/cw20-escrow/src/state.rs (hunk 2 above)
Comment (ethanfrey) [reply contracts/cw20-escrow/src/state.rs:80]: I rather keep the user-facing API in seconds.
I guess this is state, but then we expose this in query.

File: .circleci/config.yml (hunk 1 above)
Comment (ethanfrey) [reply .circleci/config.yml]: Works. Thanks for that

**Review by maurolacy — APPROVED** [review]
lgtm.

**Pull request merged**